model/postgres/postgres_server.rb

```diff
@@ -335,7 +339,28 @@ def attach_s3_policy_if_needed
     end
   end
 
+  def observe_archival_backlog(session:)
+    result = session[:ssh_session].exec!(
+      "find /dat/:version/data/pg_wal/archive_status -name '*.ready' | wc -l",
```

Comment:
I’m not entirely sure, but would `ls dat/:version/data/pg_wal/archive_status/*.ready | wc -l`
be faster? Is there any real benefit to using find here?

Also, I’ve noticed that in some cases `ls ...` returns an error saying a file was not found. I believe this is a race condition: the shell expands the glob, and then `ls` attempts to access the file’s metadata, but the file gets deleted in between, resulting in an error.

Would `find` have a similar issue? In that scenario, do `ls` or `find` return a non-zero exit code? If so, could that cause `observe_archival_backlog` to fail?

Long story short, could you please reproduce this behavior and make sure the function correctly handles this edge case?